### PR TITLE
[0.6.x] CLI and controller updates for weights and rollouts

### PR DESCRIPTION
**Pull request by cbron**
https://github.com/rancher/rio/issues/774
https://github.com/rancher/rio/issues/719

This PR does several things: 

* breaks out any tricky math into functions and unit tests them, including fixes and tests for edge cases
* fixes bug where run command wouldn't de-promote other services
* add magic so rollout is smoother
* fixed general bugs around rollout controller and weights
* updates rollout duration to 4 seconds from 2, to help avoid conflicts in api
* add note to cli and docs that rollout duration is estimation

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -43,5 +43,6 @@ if [[ ${ARCH} == amd64 ]]; then
 
     trap print_log exit
 
-    go test -v ./tests/integration/... --integration-tests
+    echo 'Running integration tests...'
+    go test ./tests/integration/... --integration-tests
```
Comment (cbron) [inline scripts/test]: I changed behavior of test output so it wasn't so cluttered

File: cli/cmd/weight/weight.go [hunk 2]
```diff
@@ -18,10 +19,13 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-const DefaultInterval = 2
+const (
+	DefaultInterval = 4
```
Comment (cbron) [inline cli/cmd/weight/weight.go]: Default interval is now 4

File: cli/cmd/create/create.go [hunk 3]
```diff
@@ -98,7 +98,22 @@ func (c *Create) RunCallback(ctx *clicontext.CLIContext, cb func(service *riov1.
 	}
 
 	service = cb(service)
-	return service, ctx.Create(service)
+	err = ctx.Create(service)
+	if err != nil {
+		return nil, err
+	}
+	if c.Weight == 100 {
+		err = weight.PromoteService(ctx, types.Resource{
```
Comment (StrongMonkey) [inline cli/cmd/create/create.go]: Nit: 
```suggestion
		return service,  weight.PromoteService(ctx, types.Resource{
```
